src/pages/accountsettings/AccountSettings.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, Component } from "react";
+import { Link } from "react-router-dom";
+// import "../signup/signup.css";
+
+import "../../common.css"
+
+const initialValues = {
+  email: "",
+  password: "",
+};
+
+function AccountSettings() {
+  const [component, setComponent] = useState("signup");
+
+  // States for registration
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [userName, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [repeatPassword, setRepeatPassword] = useState("");
+
+  // Handling the name change
+  const handleName = (e) => {
+    setName(e.target.value);
+  };
+
+  // Handling the phone number change
+  const handlePhoneNumber = (e) => {
+    setPhoneNumber(e.target.value);
+  };
+
+  // Handling the email change
+  const handleEmail = (e) => {
+    setEmail(e.target.value);
+  };
+
+  // Handling the username change
+  const handleUserName = (e) => {
+    setUsername(e.target.value);
+  };
+
+  // Handling the password change
+  const handlePassword = (e) => {
+    setPassword(e.target.value);
+  };
+
+  // Handling the repeat password change
+  const handleRepeatPassword = (e) => {
+    setRepeatPassword(e.target.value);
+  };
+
+  return (
```

Comment:
You should look into the `form` element of HTML in case you don't know it.
This form is triggered by a submit button. And if you make your inputs required, the form will be submitted as soon as everything is filled. Now, the button redirects me to your next page, without having filled a single detail in the input fields.